src/shared/src/logistics/redux/taskListEntityUtils.js

```diff
@@ -10,3 +24,70 @@ export function findTaskListByTask(taskListsById, task) {
     return _.includes(taskList.itemIds, task['@id']);
   });
 }
+
+export function addAssignedTask(taskListsById, task) {
```

Comment:
I think regarding the UI you will need to support an index parameter, as to insert task(s) in the middle of tasklists.

Reply:
Yep we'll need it for the new UI and that's why we let it as it is for now.
We'll making the needed changes once we start supporting the new UI and in that point we'll need the index.